### ISSUE TITLE
Production parameter "Tagged"

### ISSUE THREAD

**Issue by bathos** [tc39/proposal-template-literal-revision#4]
I’m curious to know more about the choice to introduce the Tagged syntactic production parameter. Existing parameterized productions are "shorthand" for 2 or more permutations that end up being meaningfully different somewhere down the line. All existing parameterized productions have in common the possibility of a descendent production that includes a member whose presence is conditional on whether that parameter is true or false. It seemed to me that the parameter notation (which is expressly called "shorthand" for these permutations, e.g. `StatementList[Return]` --> `StatementList` + `StatementList_Return`) was primarily for avoiding redundancy in the grammar specification for these kinds of cases.

This spec introduces a new parameter, but it does not introduce any productions predicated on that parameter. If I understand correctly, in this case it is only being used as a ‘carrier’ for metadata. If that’s the case, I would think that an alternative production, rather than parameterizing existing productions, would be more consistent with patterns established by the grammar.

i.e., something like this:

<img width="450" alt="screen shot 2016-10-09 at 6 20 33 pm" src="https://cloud.githubusercontent.com/assets/6257356/19224394/027078c8-8e53-11e6-8d4b-4b30c87ef0c3.png">
<img width="971" alt="screen shot 2016-10-09 at 6 23 46 pm" src="https://cloud.githubusercontent.com/assets/6257356/19224396/0615d61c-8e53-11e6-9f20-fc3b7145aa95.png">

Other productions (except the lexical ones) would not be changed. The early errors etc would be predicated on the parent production (as many already are) instead of the parameter:

<img width="517" alt="screen shot 2016-10-09 at 7 01 40 pm" src="https://cloud.githubusercontent.com/assets/6257356/19224398/0bfb4936-8e53-11e6-9e8e-7a7311d4f25a.png">

(etc). Unless I’m missing something here (a major possibility), this approach appears to get the same job done with fewer "touches" to the grammar + consistency with existing patterns.

It’s possible that using production parameters as in the current proposal is not really inconsistent with the original intent at all, and that it’s only a coincidence that currently they all lead to actual differences w/ regard to "set of possible valid token sequences".

tldr: Why the parameterized approach?

[edit: typos in screenshot]
